libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -640,6 +640,32 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			array <<
 			"))\n";
 
+			break;
+	}
+	case FunctionType::Kind::ArrayPop:
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(
+			*dynamic_cast<MemberAccess const&>(_functionCall.expression()).expression().annotation().type
+		);
+		defineExpression(_functionCall) <<
+			m_utils.storageArrayPopFunction(arrayType) <<
+			"(" <<
+			m_context.variable(_functionCall.expression()) <<
+			")\n";
+		break;
+	}
+	case FunctionType::Kind::ArrayPush:
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(
+			*dynamic_cast<MemberAccess const&>(_functionCall.expression()).expression().annotation().type
+		);
+		defineExpression(_functionCall) <<
+			m_utils.storageArrayPushFunction(arrayType) <<
+			"(" <<
+			m_context.variable(_functionCall.expression()) <<
+			", " <<
+			(arguments.empty() ? m_utils.zeroValueFunction(*arrayType.baseType()) + "()" : m_context.variable(*arguments.front())) <<
```

Comment:
No type conversion?

Reply:
Actually I think, we should use `expressionAsType` instead of `m_context.variable`, unless this is about an lvalue or an "arbitrary type" thing.

Reply:
@chriseth Changed it such that `expressionAsType` is being used instead.